### PR TITLE
Feature/start es lib build

### PR DESCRIPTION
**Pull request by AndyOGo**
improves #332 

Changes proposed in this pull request:

 - start adding a library build for **ES Modules** format
 - adds package.json module flag
 - extends release script and build script
- makes `withReact` JSX-free

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
 
 # How Has This Been Tested?
 
 
 
 # Checklist:
 
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my own code
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation
 - [ ] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LucaMele** [comment]
fixes #430

**Comment by LucaMele** [comment]
LGTM but didnt test it locally

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "2.0.1-beta.128",
   "description": "AXA Switzerland Web Patterns",
   "main": "stack/server.js",
+  "module": "lib",
```
Comment (AndyOGo) [inline package.json]: Guess this expect a single module, not a folder containing multiple modules 🤔 
https://github.com/rollup/rollup/wiki/pkg.module

Comment (AndyOGo) [reply package.json]: https://github.com/rollup/rollup/issues/1593

**Pull request merged**